scripts/gulp/utils/childProcessUtils.ts

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable no-console */
+import { exec, ExecOptions, spawn, SpawnOptions } from 'child_process'
+import through2 from 'through2'
+import util from 'util'
+
+// import psTree from 'ps-tree'
+// import psNode from 'ps-node'
+// import util from 'util'
+
+import { prefixStream } from './prefixStream'
+
+const spawningApps = new Set()
+// const killAsync = util.promisify(psNode.kill)
+// const psTreeAsync = util.promisify(psTree)
+// const runningApps = new Map<
+//   AllSpawnableApps,
+//   [ChildProcess, ArgsFor<typeof spawned>, Function]
+// >()
+
+export async function allReady () {
+  while (spawningApps.size > 0) {
+    await new Promise((ready) => setTimeout(ready, 100))
+  }
+
+  return true
+}
+
+export type AllSpawnableApps =
+  | `vite-${string}`
+  | `vite:build-${string}`
+  | 'gql-codegen'
```

Comment:
Is this valid code? Github isn't syntax highlighting it because of the template strings.

Reply:
Yep. This is https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html - newer feature, GH is probably lagging 😄